src/hotspot/share/memory/arena.hpp

```diff
@@ -151,9 +150,11 @@ class Arena : public CHeapObj<mtNone> {
       return old;
     }
   }
-  // Further assume size is padded out to words
+
+  // Allocate in the arena, assuming the size has been aligned to size of pointer, which
+  // is 4 bytes on 32 bits, hence the name.
```

Comment:
`Amalloc_4` is a horrible name for this, because this is allocating pointer-aligned and sized values, and even enforces that the size is appropriately pointer aligned.  Maybe better would be something like `AmallocP`?  Though it can also be used to allocate `size_t` and the like that are non-pointers.

Reply:
I agree that Amalloc_4 is a bad name.  I thought of Amalloc_ptr but it's not good either.  Amalloc_naturally_aligned....  too long.

Reply:
I agree. I feel like this should really be the default "Amalloc" (pointer-sized alignment) and Amalloc should really be specific, e.g. "Amalloc64".

Reply:
For some reason, the new me doesn't want to change all of these calls, but there are _only_ 45 Amalloc and now 38 Amalloc_4s.
I agree with both of you though.  Amalloc should be the default pointer sized alignment, and existing Amalloc should be jlong aligned.  So Amalloc_4 => Amalloc and Amalloc => Amalloc64.  I suspect the Amalloc64s could really be Amalloc but that only matters on 32 bit platforms which we don't support.  They can be changed by supporters of 32 bit platforms though.

Reply:
It seems like this can do that renaming to get this done.  The reason I worked on this enhancement is to get it out of the backlog, so I don't want to add another fairly simple RFE to the backlog.

Reply:
I filed JDK-8270179 for the follow-on renaming.  It's almost trivial but not quite.